dsweb/src/main/java/com/doctusoft/dsw/client/gwt/BarChartRenderer.java

```diff
@@ -59,11 +66,19 @@ protected void changed() {
 				}
 				initializeComponent();
 			}
-			
+
 		};
-		
+
+	}
+
+	/**
+	 * Rerender plot container if the model does not have id
+	 */
+	private void initializeId() {
+		chartId = Objects.firstNonNull(model.getId(), "bar_chart_" + idCounter++);
```

Comment:
@mterjeki are you sure this works properly? The id assigned to the div when invoking the super constructor already increments the idCounter. I'd suggest that you don't specify the id on the super constructor call, but you assign it in the initializeId method using ```widget.attr("id", chartId)```